### PR TITLE
fix olm rbac and operator-tls OwnerReferences

### PR DESCRIPTION
**Pull request by dkhachyan**

### CODE REVIEW

File: pkg/controller/cluster/operator.go [hunk 1]
```diff
@@ -176,9 +176,9 @@ func (c *Controller) createOperatorSecret(ctx context.Context, operator metav1.O
 			Labels:    labels,
 			OwnerReferences: []metav1.OwnerReference{
 				*metav1.NewControllerRef(operator, schema.GroupVersionKind{
-					Group:   miniov2.SchemeGroupVersion.Group,
-					Version: miniov2.SchemeGroupVersion.Version,
-					Kind:    miniov2.OperatorCRDResourceKind,
+					Group:   "apps",
+					Version: "v1",
+					Kind:    "Deployment",
```
[the remark this answers is not in the document]
Comment (harshavardhana) [reply pkg/controller/cluster/operator.go:181]: @Alevsk ^^

[the remark this answers is not in the document]
Comment (Alevsk) [reply pkg/controller/cluster/operator.go:181]: @harshavardhana we removed the owner reference in the latest release, its not need it, CSRs are removed automatically every 15 mins

**Comment by harshavardhana** [comment]
Please test with latest release v4.2.7 and let us know if we need this PR @dkhachyan

**Comment by dkhachyan** [comment]
I`v tried 4.2.7, it has the same problem. It is not enough to remove the owner from csr it is also necessary to fix owner of the secret

**Comment by harshavardhana** [comment]
> I`v tried 4.2.7, it has the same problem. It is not enough to remove the owner from csr it is also necessary to fix owner of the secret

So this PR is needed in it's entirety ?

// cc @Alevsk

**Review by dvaldivia — APPROVED** [review]

**Review by dvaldivia — APPROVED** [review]

**Comment by dvaldivia** [comment]
@dkhachyan can you rebase your PR?

**Comment by dvaldivia** [comment]
Merged via https://github.com/minio/operator/pull/832

**Pull request closed without merging**